### PR TITLE
[README] Add rebasing example

### PR DESCRIPTION
**Pull request by Abdelrahman-Kh-Fouad**

### CODE REVIEW

File: readme_utils/rebasing.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+You should rebasing your branch to rearrange the commits of your branch.
```
Comment (certik) [inline readme_utils/rebasing.md:1]:
```suggestion
You should be rebasing your branch to rearrange the commits of your branch.
```

**Comment by certik** [comment]
Excellent, thank you!

**Review by czgdp1807 — COMMENTED** [review]
I don’t use this approach for cleaning up my git history. I follow the steps in https://github.com/lcompilers/lpython/pull/783#issuecomment-1188875210. Squashing using interactive rebase sometimes leads to messing up of git history so I do it only when its outcome is predictable for me.

**Comment by namannimmo10** [comment]
Yep. Maybe we can add another line here, mentioning: "create a backup branch before doing this interactive rebase."

**Comment by certik** [comment]
We should put both approaches there, I use both.

@czgdp1807's approach is what I use for a complicated history with lots of merges, since a rebase will give me conflicts, but the `git reset master` approach is solid.

File: readme_utils/rebasing.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+Assuming initial state is your current branch (say `xyz_branch`),
+
+1. git checkout main or git checkout master (whichever is being used in a certain project as the lead branch).
+2. git pull origin main (origin is the project remote say for lpython it will be pointing to (https://github.com/lcompilers/lpython).
+3. `git checkout xyz_branch`
+4. `git merge main` or `git merge master`
+5. `git reset main` or `git reset master`
+6. `git add xyz_file`
+7. Repeat step 6 until you are satisfied with the group of changes you want to commit.
+8. `git commit -m "nice_commit_message"` or `git commit (and then write detailed commit message n the command line editor)`.
```
Comment (czgdp1807) [inline readme_utils/rebasing.md:244]:
```suggestion
8. `git commit -m "nice_commit_message"` or `git commit (and then write detailed commit message in the command line editor)`.
```

**Review by certik — APPROVED** [review]
I think this is a good start, it has all the information. I think the presentation can be more polished, but we can improve upon it with subsequent PRs I think.

**Pull request merged**